packages/react/src/components/StructuredList/StructuredList.js

```diff
@@ -176,40 +190,52 @@ StructuredListRow.propTypes = {
   /**
    * Specify whether a `<label>` should be used
    */
-  label: PropTypes.bool,
+  label: deprecate(
+    PropTypes.bool,
+    `\nThe prop \`label\` will be removed in the next major version of Carbon.`
+  ),
 
   /**
    * Provide a handler that is invoked on the key down event for the control,
-   * if `<label>` is in use
    */
   onKeyDown: PropTypes.func,
-
-  /**
-   * Specify the tab index of the container node, if `<label>` is in use
-   */
-  tabIndex: PropTypes.number,
 };
 
 StructuredListRow.defaultProps = {
   head: false,
-  label: false,
-  tabIndex: 0,
   onKeyDown: () => {},
 };
 
 export function StructuredListInput(props) {
-  const { className, value, name, title, id, ...other } = props;
-  const classes = classNames(`${prefix}--structured-list-input`, className);
-  const instanceId = id || getInstanceId();
+  const defaultId = useId('structureListInput');
+  const {
+    className,
+    name = `structured-list-input-${defaultId}`,
+    title,
+    id,
+    ...other
+  } = props;
+  const classes = classNames(
+    `${prefix}--structured-list-input`,
+    `${prefix}--visually-hidden`,
+    className
+  );
+  const row = React.useContext(GridRowContext);
+  const selectedRow = React.useContext(GridSelectedRowStateContext);
+  const setSelectedRow = React.useContext(GridSelectedRowDispatchContext);
 
   return (
     <input
       {...other}
       type="radio"
-      tabIndex={-1}
-      id={instanceId}
+      tabIndex={0}
+      checked={row && row.id === selectedRow}
+      value={row ? row.id : ''}
```

Comment:
Something I'm a little unsure about here is if they still pass in a `value` what should we do with it?

One tricky thing with adding in context is that if, for whatever reason, someone uses `StructuredListInput` without the wrapper components then this component ends up not working because the context is never defined.

I guess this kind of goes back to your point earlier about how this might be too big of a breaking change to safely migrate to in v10? Curious what your thoughts are

Reply:
its worth talking about for sure -- I guess what are the chances or what is the actual use case of _only using_ StructuredListInput? Should we put a call out on Slack or maybe kick this whole can down the road to v11?

Reply:
@dakahn maybe the safest thing is just to wrap it all behind a flag, I guess? If we want to do it in v10, then I think we'll just need to make sure we understand what props teams could be supplying and how to leverage/use them, if so.

For example, if someone is using a `label` then they might be expecting in a test to be able to find a node using that `label`. If we no longer use it in the component, then their test would be break.